nixos/modules/services/web-apps/dawarich.nix

```diff
@@ -0,0 +1,685 @@
+        };
+
+        port = lib.mkOption {
+          description = "The port of the redis server Dawarich will connect to.";
+          type = lib.types.port;
+          default = 0;
+        };
+      };
+
+      database = {
+        createLocally = lib.mkOption {
+          description = ''
+            Whether to configure a local PostgreSQL server and database for Dawarich.
+            The connection is performed via Unix sockets.
+          '';
+          type = lib.types.bool;
+          default = true;
+        };
+
+        host = lib.mkOption {
+          type = lib.types.str;
+          default = "/run/postgresql";
+          example = "127.0.0.1";
+          description = "Hostname or address of the postgresql server. If an absolute path is given here, it will be interpreted as a unix socket path.";
+        };
+
+        port = lib.mkOption {
+          type = lib.types.nullOr lib.types.port;
+          default = 5432;
+          description = "Port of the postgresql server.";
+        };
+
+        name = lib.mkOption {
+          type = lib.types.str;
+          default = "dawarich";
+          description = "The name of the dawarich database.";
+        };
+
+        user = lib.mkOption {
+          type = lib.types.str;
+          default = "dawarich";
+          description = "The database user for dawarich.";
+        };
+
+        passwordFile = lib.mkOption {
+          type = lib.types.nullOr lib.types.path;
+          default = null;
+          example = "/run/keys/dawarich-db-password";
+          description = ''
+            A file containing the password corresponding to {option}`${opt.database.user}`.
+          '';
+        };
+      };
+
+      smtp = {
+        host = lib.mkOption {
+          type = lib.types.nullOr lib.types.str;
+          default = null;
+          description = "SMTP host used when sending emails to users.";
+        };
+
+        port = lib.mkOption {
+          type = lib.types.port;
+          default = 25;
+          description = "SMTP port used when sending emails to users.";
+        };
+
+        fromAddress = lib.mkOption {
+          type = lib.types.nullOr lib.types.str;
+          default = null;
+          example = "dawarich@example.com";
+          description = ''"From" address used when sending emails to users.'';
+        };
+
+        user = lib.mkOption {
+          type = lib.types.nullOr lib.types.str;
+          default = null;
+          example = "dawarich@example.com";
+          description = "SMTP login name.";
+        };
+
+        passwordFile = lib.mkOption {
+          type = lib.types.nullOr lib.types.path;
+          default = null;
+          example = "/run/keys/dawarich-smtp-password";
+          description = ''
+            Path to file containing the SMTP password.
+          '';
+        };
+      };
+
+      package = lib.mkPackageOption pkgs "dawarich" { };
+
+      extraConfig = lib.mkOption {
+        type = lib.types.attrs;
```

Comment:
Should be `attrsOf str` or `attrsOf (nullOr (oneOf [ str path package ])`. https://github.com/NixOS/nixpkgs/blob/2ec33b1c5c862b67e9f1115cbcc41983d44a2f88/nixos/lib/systemd-unit-options.nix#L352-L358